src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2469,12 +2479,12 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                         return null;
                     }
 
-                    // Signing can only be done to on-disk files. This is a limitation of the CLR APIs which we use 
-                    // to perform strong naming. If this binary is configured to be signed, create a temp file, output to that
+                    // If the current strong name provider is the Desktop version, signing can only be done to on-disk files. 
```

Comment:
This method has gotten pretty complicated. Consider extracting the lambda here into a separate method or some other refactoring.